lib/active_record/pt_osc_migration.rb

```diff
@@ -2,7 +2,19 @@
 require 'active_record/connection_adapters/mysql_pt_osc_adapter'
 require 'shellwords'
 
+module MigratorExtensions
+  def record_version_state_after_migrating(version)
+    ActiveRecord::Base.logger.debug 'Verifying active connections prior to recording version' if ActiveRecord::Base.logger
+    ActiveRecord::Base.verify_active_connections! # Reconnect to DB if it's gone away while we were migrating.
+    super
+  end
+end
+
 module ActiveRecord
+  class Migrator
+    prepend MigratorExtensions
+  end
```

Comment:
[This doc](https://www.ruby-lang.org/en/news/2014/01/10/ruby-1-9-3-will-end-on-2015/)
suggested that even security patches haven't been made to 1.9.3 for over a year at this point, so it seemed safe to deprecate?